src/test/java/web/tests/FlipkartTest.java

```diff
@@ -166,4 +170,30 @@ public void testAddToCartFunctionality(String product) {
         }
         softAssert.assertThat(Integer.parseInt(cartPage.getShoePriceTotalInCart()) == totalPriceOfShoeSelectedFromList).isTrue();
     }
+
+    /**
+     * Below testcase is clicking policies and checking the header and back to top button
+     */
+    @Test
+    public void testPolicyBackToTop() {
+        ArrayList<String> policyElementsWhichNeedToClick = new ArrayList<>(Arrays.asList(map.get("policyElementsWhichNeedToClick").split(",")));
+        ArrayList<String> policyHeaders = new ArrayList<>(Arrays.asList(map.get("policyHeaders").split(",")));
+
+        HomePage homePage = new HomePage().popUpCancel();
+
+        for (int i = 0; i < policyHeaders.size(); i++) {
+            System.out.println("policyElementsWhichNeedToClick = " + policyElementsWhichNeedToClick.get(i));
+            PolicySubPage policySubPage = homePage.clickPolicySingleElement(policyElementsWhichNeedToClick.get(i));
+
+            softAssert.assertThat(policySubPage.isHeaderDisplayed(policyHeaders.get(i))).as("Header Not Displayed in = " + policyHeaders.get(i)).isTrue();
+
+            softAssert.assertThat(policySubPage.verifyBackToTopIsDisplayedAndClickButton(policyElementsWhichNeedToClick.get(i))).as("verify back To top is not displayed = " + policyElementsWhichNeedToClick.get(i)).isTrue();
+            softAssert.assertThat(policySubPage.checkIfBackToTopButtonIsDisapper(policyElementsWhichNeedToClick.get(i))).as("Back to top button is not disappered after clicking in " + policyHeaders.get(i)).isFalse();
+
+            softAssert.assertThat(policySubPage.verifyPageGoesUp(policyElementsWhichNeedToClick.get(i))).as("Page = " + policyHeaders.get(i) + " doesnt scrolled up").isTrue();
+            softAssert.assertThat(policySubPage.isHeaderDisplayed(policyHeaders.get(i))).as("After clicking back to top button, Header Not Displayed in = " + policyHeaders.get(i)).isTrue();
```

Comment:
Can we break each line into a 2 lines - to enhance readability.
Also the design looks messed up, need to review